### ISSUE TITLE
A phpDoc template closing tag is mistakenly considered belonging to the following function

### ISSUE THREAD

**Issue by morozov** [squizlabs/PHP_CodeSniffer#2838]
Some sniffs from the "Squiz" collection are used in [doctrine/coding-standard](https://github.com/doctrine/coding-standard/blob/931c039c767e746420f24ac6fa11ed11aac3e7b8/lib/Doctrine/ruleset.xml#L436). The standard does not enforce a phpDoc block for each symbol and, furthermore, forbids the usage of useless doc blocks.

Consider the code example:
```php
<?php

class Test
{
    /**#@+
     * Error codes.
     */
    const ERR_1 = 1;
    const ERR_2 = 1;
    /**#@-*/

    private function doStuff() : void
    {
    }
}
```
If a phpDoc template ([example](https://github.com/pear/pear-core/blob/9e9014d4aa662db336eef3121011bdcd27f5e63f/PEAR.php#L21-L34)) is followed by a method without a docblock, then the "Squiz.Commenting.FunctionComment" sniff mistakenly attributes the closing template block to the method and marks it as invalid:
```
phpcs --standard=Squiz Test.inc -s

FILE: /home/morozov/Projects/dbal/Test.inc
------------------------------------------------------------------------------------------------
FOUND 14 ERRORS AFFECTING 7 LINES
------------------------------------------------------------------------------------------------
...
 10 | ERROR | [ ] There must be no blank lines after the function comment
    |       |     (Squiz.Commenting.FunctionComment.SpacingAfter)
...
------------------------------------------------------------------------------------------------
```

**Comment by gsherwood** [comment]
Docblock templates are completely unsupported by the Squiz coding standard as they are never used within Squiz code. Adding proper support would be more than just ignoring the closing comment here, and it would certainly generate a lot more errors that you wont want. The Squiz standard tries to enforce a docblock for everything, so any changes that move it away from that direction would not be considered.

What you really need is a generic collection of docblock sniffs that are flexible enough to implement your own rules, but it's not something that is being worked on.

**Comment by morozov** [comment]
> What you really need is a generic collection of docblock sniffs that are flexible enough to implement your own rules, but it's not something that is being worked on.

Agree. I believe that this is the idea behind [slevomat/coding-standard](https://github.com/slevomat/coding-standard). Instead of reusing code from the "Squiz" standard, maybe there's a better way to implement the same rules.

**Comment by jrfnl** [comment]
> What you really need is a generic collection of docblock sniffs that are flexible enough to implement your own rules, but it's not something that is being worked on.

Actually, this is being worked on, just not yet public ;-)
